services/auth/reverseproxy.go

```diff
@@ -39,7 +39,12 @@ type ReverseProxy struct{}
 func (r *ReverseProxy) getUserName(req *http.Request) string {
 	webAuthUser := strings.TrimSpace(req.Header.Get(setting.ReverseProxyAuthUser))
 	if len(webAuthUser) == 0 {
-		return ""
+		email := strings.TrimSpace(req.Header.Get(setting.ReverseProxyAuthEmail))
+		if len(email) == 0 {
+			return ""
+		}
+		webAuthUser := strings.Split(email, "@")[0]
```

Comment:
If `email` doesn't contain `@`, we need some handler.

Reply:
`if !strings.Contains(email, "@") {
			return ""
		}
		webAuthUser := strings.Split(email, "@")[0]`

@lunny we can return null if email doesn't have a `@` Let me know your thoughts.

Reply:
If `@` is the first char of that string?

Reply:
@lunny  - 

```
// getUserName extracts the username from the "setting.ReverseProxyAuthUser" header
func (r *ReverseProxy) getUserName(req *http.Request) string {
	webAuthUser := strings.TrimSpace(req.Header.Get(setting.ReverseProxyAuthUser))
	if len(webAuthUser) == 0 {
		email := strings.TrimSpace(req.Header.Get(setting.ReverseProxyAuthEmail))
		if !strings.Contains(email, "@") {
			log.Trace("Not a valid email address %v", email)
			return nil
		} else if strings.HasPrefix(email, "@") {
			log.Fatal("Invalid email address found : %v", err)
		} else {
			webAuthUser := strings.Split(email, "@")[0]
			return webAuthUser
		}
	}
	return webAuthUser
}
```
We can add something like above to check for first character.

Reply:
Created a PR with latest changes https://github.com/go-gitea/gitea/pull/21264